bot/cogs/moderation/infractions.py

```diff
@@ -225,7 +225,10 @@ def __init__(self, bot: Bot):
 
         self.mod_log.ignore(Event.member_remove, user.id)
 
-        action = user.kick(reason=reason)
+        if len(reason) > 512:
+            log.info("Kick reason is longer than 512 characters. Reason will be truncated for Audit Log.")
+
+        action = user.kick(reason=f"{reason[:509]}..." if len(reason) > 512 else reason)
```

Comment:
This can add a `...` to the middle of a word, like "See you ne..." - You will want to truncate at whitespace, `textwrap` will be of use here.